### PR TITLE
enforced lockfile for build-and-lint-test

### LINKED ISSUE #14610: Pin build-and-lint dependencies
Following up on #13876: the badly-named "`build-and-lint`" integration test between the JSON API and the TS bindings currently uses unpinned dependencies. That's a bit of a ticking bomb that is going to explode in our face at some point, as it did last May (when #13876 was made to patch a specific dependency resolution path); the proper fix would be to commit and reuse a specific lockfile for those tests, ensuring that without manual intervention they keep running with the exact same set of dependencies.

Some investigative work is needed to figure out exactly where to find, and where to store, that lockfile.

Initially assigned to @ray-roestenburg-da and @stefanobaghino-da for prioritization.

### PR DESCRIPTION
**Pull request by S11001001**
Fixes #14610.

* [ ] ensure that package.json/yarn.lock mismatch causes build failure
* [ ] yarn audit
* [ ] update lockfile
* [ ] no changelog

### CODE REVIEW

File: language-support/ts/codegen/tests/ts/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,3614 @@
...
+"@daml.js/40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7@file:daml2js/40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7":
+  version "0.0.0"
+  dependencies:
+    "@mojotech/json-type-validation" "^3.1.0"
+
+"@daml.js/build-and-lint-1.0.0@file:daml2js/build-and-lint-1.0.0":
+  version "0.0.0"
+  dependencies:
+    "@daml.js/40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7" "file:../../../../../../../var/folders/hd/0fcpzy1173dgvs6k2vdq77kw0000gp/T/tmp.E6R77CVsdM/yarn/v6/npm-@daml-js-build-and-lint-1-0-0-0.0.0-d13c265f-103a-42a0-ac7b-ee0b94e3b512-1661873860741/node_modules/@daml.js/40f452260bef3f29dede136108fc08a88d5a5250310281067087da6f0baddff7"
+    "@daml.js/d14e08374fc7197d6a0de468c968ae8ba3aadbf9315476fd39071831f5923662" "file:../../../../../../../var/folders/hd/0fcpzy1173dgvs6k2vdq77kw0000gp/T/tmp.E6R77CVsdM/yarn/v6/npm-@daml-js-build-and-lint-1-0-0-0.0.0-d13c265f-103a-42a0-ac7b-ee0b94e3b512-1661873860741/node_modules/@daml.js/d14e08374fc7197d6a0de468c968ae8ba3aadbf9315476fd39071831f5923662"
```
Comment (S11001001) [inline language-support/ts/codegen/tests/ts/yarn.lock:302]: Need to figure out how to exclude these from the lockfile.

File: language-support/ts/codegen/tests/build-and-lint.sh [hunk 2]
```diff
@@ -67,7 +67,15 @@ PATH=$PATH:$GRPCURL
 
 # Build, lint, test.
 cd build-and-lint-test
-$YARN install --pure-lockfile > /dev/null
+$YARN install > /dev/null
+# simulating what yarn install --frozen-lockfile is supposed to do,
+# because --frozen-lockfile appears to behave exactly like
+# --pure-lockfile - #14873
+if ! /usr/bin/diff -du $TS_DIR/yarn.lock $TMP_DIR/yarn.lock; then
```
Comment (S11001001) [inline language-support/ts/codegen/tests/build-and-lint.sh:74]: Also don't want to have `/usr/bin/diff` hardcoded here.

File: language-support/ts/codegen/tests/ts/yarn.lock (hunk 1 above)
Comment (S11001001) [reply language-support/ts/codegen/tests/ts/yarn.lock:302]: Or exclude them from the diff by removing declarations from both files.

Comment (S11001001) [reply language-support/ts/codegen/tests/ts/yarn.lock]: Used sed.

**Review by ray-roestenburg-da — APPROVED** [review]

**Pull request closed without merging**